core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala

```diff
@@ -34,6 +34,23 @@ class WorkerSuite extends SparkFunSuite with Matchers {
   }
   def conf(opts: (String, String)*): SparkConf = new SparkConf(loadDefaults = false).setAll(opts)
 
+  private var _worker: Worker = _
+  private def makeWorker(conf: SparkConf): Worker = {
+    assert(_worker === null, "Some Worker's RpcEnv is leaked in tests")
+    val securityMgr = new SecurityManager(conf)
+    val rpcEnv = RpcEnv.create("test", "localhost", 12345, conf, securityMgr)
+    _worker = new Worker(rpcEnv, 50000, 20, 1234 * 5, Array.fill(1)(RpcAddress("1.2.3.4", 1234)),
+      "Worker", "/tmp", conf, securityMgr)
+    _worker
+  }
+  after {
```

Comment:
That's bad practice, and why `override` exists. For example if you wrote `afer {` you'd maybe never notice that your cleanup code was silently not being triggered.

The code doesn't do this consistently, but more often does write the full declaration, which is probably a good idea. I'd recommend changing this one.